types/lodash/common/lang.d.ts

```diff
@@ -574,6 +574,7 @@ declare module "../index" {
         isEmpty<T extends { __trapAny: any }>(value?: T): boolean;
         isEmpty(value: string): value is '';
         isEmpty(value: Map<any, any> | Set<any> | List<any> | null | undefined): boolean;
+        isEmpty(value: object): boolean;
```

Comment:
Yeah I think lower complexity is better unless the complexity is giving a clear win. For an edge case like `isEmpty({})` I don't think it's worth the extra complexity, so I like what you have.